packages/network/test/fakes.js

```diff
@@ -0,0 +1,129 @@
+        lp: undefined,
+        nonce: 0,
+      };
+    },
+    {
+      async onInstantiate(_port, _localAddr, _remote, _protocol) {
+        return '';
+      },
+      async onCreate(_protocol, _impl) {
+        log('created', _protocol, _impl);
+      },
+      async generatePortID() {
+        this.state.nonce += 1;
+        return `port-${this.state.nonce}`;
+      },
+      async onBind(port, localAddr) {
+        assert(port, `port is supplied to onBind`);
+        assert(localAddr, `local address is supplied to onBind`);
+      },
+      async onConnect(port, localAddr, remoteAddr) {
+        assert(port, `port is tracked in onConnect`);
+        assert(localAddr, `local address is supplied to onConnect`);
+        assert(remoteAddr, `remote address is supplied to onConnect`);
+        if (!this.state.lp) {
+          return { handler: makeEchoConnectionHandler().handler };
+        }
+        assert(this.state.l);
+        const ch = await when(
+          this.state.l.onAccept(
+            this.state.lp,
+            localAddr,
+            remoteAddr,
+            this.state.l,
+          ),
+        );
+        return { localAddr, handler: ch };
+      },
+      async onListen(port, localAddr, listenHandler) {
+        assert(port, `port is tracked in onListen`);
+        assert(localAddr, `local address is supplied to onListen`);
+        assert(listenHandler, `listen handler is tracked in onListen`);
+        this.state.lp = port;
+        this.state.l = listenHandler;
+        log('listening', port.getLocalAddress(), listenHandler);
+      },
+      async onListenRemove(port, localAddr, listenHandler) {
+        assert(port, `port is tracked in onListen`);
+        assert(localAddr, `local address is supplied to onListen`);
+        assert.equal(
+          listenHandler,
+          this.state.lp && this.state.l,
+          `listenHandler is tracked in onListenRemove`,
+        );
+        this.state.lp = undefined;
+        log('port done listening', port.getLocalAddress());
+      },
+      async onRevoke(port, localAddr) {
+        assert(port, `port is tracked in onRevoke`);
+        assert(localAddr, `local address is supplied to onRevoke`);
+        log('port done revoking', port.getLocalAddress());
+      },
+    },
+  );
+
+  return makeProtocolHandler;
+};
+
+/**
+ *
+ * @param {Zone} zone
+ */
+export const fakeNetworkEchoStuff = zone => {
+  const powers = prepareVowTools(zone);
+  const { makeVowKit, when } = powers;
+
+  const makeNetworkProtocol = prepareNetworkProtocol(zone, powers);
+  const makeEchoConnectionHandler = prepareEchoConnectionKit(zone);
+  const makeProtocolHandler = prepareProtocolHandler(
+    zone,
+    makeEchoConnectionHandler,
+    powers,
+  );
+  const protocol = makeNetworkProtocol(makeProtocolHandler());
+
+  const makePortAllocator = preparePortAllocator(zone, powers);
+  const portAllocator = makePortAllocator({ protocol });
+
+  return {
+    makeEchoConnectionHandler,
+    makeVowKit,
```

Comment:
Maybe it makes sense to return `power` here? This will also make it easier to leverage `watch` if we need it in the tests

Reply:
I figured when it's needed it can be added. This is just test code so it's easy and quick to change.